apis_core/generic/forms/__init__.py

```diff
@@ -40,12 +42,23 @@ class GenericFilterSetForm(forms.Form):
 
     def __init__(self, *args, **kwargs):
         model = kwargs.pop("model")
+        annotations = kwargs.pop("annotations", {})
         super().__init__(*args, **kwargs)
         self.fields["columns"].choices = [
             (field.name, field.verbose_name)
             for field in model._meta.fields
             if field.name not in self.columns_exclude
+        ] + [
+            (key, key) for key in annotations.keys() if key not in self.columns_exclude
         ]
+        if not self.is_bound:
+            table_modules = module_paths(model, path="tables", suffix="Table")
+            table_class = first_member_match(table_modules, GenericTable)
+            self.fields["columns"].initial = [
+                field
+                for field in table_class.Meta.fields
+                if field not in self.columns_exclude
+            ]
```

Comment:
This part makes me unhappy, but I also don't know any other way to access the table fields. The main downside is, that is only gets a subset of fields - I can add fields in the `__init__` of the table, I can hide fields in the table before it is being rendered ...